internal/bminventory/inventory.go

```diff
@@ -4230,6 +4230,9 @@ func (b *bareMetalInventory) UpdateHostApprovedInternal(ctx context.Context, clu
 		log.WithError(err).Errorf("failed to update 'approved' in host: %s", hostId)
 		return err
 	}
+	hId := strfmt.UUID(hostId)
+	b.eventsHandler.AddEvent(ctx, strfmt.UUID(clusterId), &hId, models.EventSeverityInfo,
+		fmt.Sprintf("Updated Approved to %t Host %s Cluster %s", approved, hostId, clusterId), time.Now())
```

Comment:
Not related to this PR: what can we do to emit these events also from BMAC? 

/cc @filanov @avishayt

Reply:
Depend on the need. 
In agent/cluster we have a monitor in the backend that change the state of the agent/cluster so we would like to get notification about it in the resource. 
Regarding BMAC it already watch the agent CRD, it doesn't need to watch anything in the backend
In addition agent doesn't know about BMAC so it should not publish events for it, At some point BMAC will not be part of assisted service so we will need a different implementation for those events.

Reply:
Agree, we should have it in the agent not BMAC

Reply:
I don't think I explained my thoughts properly. My suggestion is not to put it in BMAC. If anything, I would like to have the updates to the Agent spec out of BMAC so that we can have all these events managed by the agent controller itself.

Right now BMAC sets the `Approve` to `true` as well as other values. 

https://github.com/openshift/assisted-service/blob/35f8de874c50871610310ad457c24017d583b6ec/internal/controller/controllers/bmh_agent_controller.go#L210

This bypasses all the events, triggers, messages, etc that are currently managed by the API, the agent controller, and other parts. I would like to solve this issue and find a solution that will work better for when BMAC would be in a separate repo.

Does the above make sense?

Reply:
it doesn't bypass anything. when agent is reconciled after you set the spec, reconcile calls backend APIs that trigger the events.